gadgets/bpfstats/README.mdx

```diff
@@ -0,0 +1,314 @@
+
+<Tabs groupId="env">
+    <TabItem value="kubectl-gadget" label="kubectl gadget">
+        ```bash
+        WIP: Headless mode for kubectl gadget is under development
+        ```
+    </TabItem>
+
+    <TabItem value="ig" label="ig">
+        Unless you configured the metrics listener to do differently, the
+        metrics will be available at `http://localhost:2224/metrics` on the
+        server side. For the `bpfstats` gadget we ran above, you
+        can find the metrics under the `bpfstats` scope:
+
+        ```bash
+        $ curl http://localhost:2224/metrics -s  | grep bpfstats
+        mapCount{gadgetImage="trace_exec:latest",gadgetName="mytraceexec",otel_scope_name="bpfstats",otel_scope_version="",progName=""} 7
+        mapCount{gadgetImage="trace_open:latest",gadgetName="mytraceopen",otel_scope_name="bpfstats",otel_scope_version="",progName=""} 6
+        mapMemory{gadgetImage="trace_exec:latest",gadgetName="mytraceexec",otel_scope_name="bpfstats",otel_scope_version="",progName=""} 1.83170336e+08
+        mapMemory{gadgetImage="trace_open:latest",gadgetName="mytraceopen",otel_scope_name="bpfstats",otel_scope_version="",progName=""} 1.71024e+06
+        otel_scope_info{otel_scope_name="bpfstats",otel_scope_version=""} 1
+        runcount_total{gadgetImage="trace_exec:latest",gadgetName="mytraceexec",otel_scope_name="bpfstats",otel_scope_version="",progName=""} 636
+        runcount_total{gadgetImage="trace_open:latest",gadgetName="mytraceopen",otel_scope_name="bpfstats",otel_scope_version="",progName=""} 219513
+        runtime_total{gadgetImage="trace_exec:latest",gadgetName="mytraceexec",otel_scope_name="bpfstats",otel_scope_version="",progName=""} 445114
+        runtime_total{gadgetImage="trace_open:latest",gadgetName="mytraceopen",otel_scope_name="bpfstats",otel_scope_version="",progName=""} 7.3383934e+07
+        ```
+    </TabItem>
+</Tabs>
+
+Finally, stop metrics collection:
+
+<Tabs groupId="env">
+    <TabItem value="kubectl-gadget" label="kubectl gadget">
+        ```bash
+        WIP: Headless mode for kubectl gadget is under development
+        ```
+    </TabItem>
+
+    <TabItem value="ig" label="ig">
+        ```bash
+        $ sudo gadgetctl delete 8423cd8e53339c8d4501ec7cdff436bc
+        8423cd8e53339c8d4501ec7cdff436bc
+        ```
+    </TabItem>
+</Tabs>
+
+### Limitations
+
+#### Memory usage of maps
+
+The shown value for MapMemory is read from `/proc/<pid>/fdinfo/<map_id>`. This
+is the maximum size the map can have, but it doesn't necessarily reflect its
+current memory allocation. Additionally, maps can be used by more than one
+program and would account towards the MapMemory of all those programs.
+
+Also note:
+* BPF_MAP_TYPE_PERF_EVENT_ARRAY: value_size is not counting the ring buffers,
+  but only their file descriptors (i.e. sizeof(int) = 4 bytes)
+* BPF_MAP_TYPE_{HASH,ARRAY}_OF_MAPS: value_size is not counting the inner maps,
+  but only their file descriptors (i.e. sizeof(int) = 4 bytes)
```

Comment:
Is it something IG-specific or it's coming from the kenel functionality itself? If it's the second, it would be great if we can add a link to the kernel documentation (if it ever exists).

Reply:
This is coming from the kernel AFAIU. TBH I [took](https://github.com/inspektor-gadget/inspektor-gadget/blob/c1acdfeed6936a957f2e4e4eef153da677d0297f/docs/gadgets/builtin/top/ebpf.md#L98-L106) this from the top ebpf which this implementation is based on. I don't think there is any kernel documentation that explains it.

Reply:
@flyth do you remember where did you take [that](https://github.com/inspektor-gadget/inspektor-gadget/pull/876) info from?